taskcluster/client.py

```diff
@@ -67,6 +74,17 @@ def __init__(self, options=None, session=None):
                     except:
                         s = '%s (%s) must be unicode encodable' % (x, credentials[x])
                         raise exceptions.TaskclusterAuthFailure(s)
+
+        baseUrl = o.get('baseUrl')
+        rootUrl = o.get('rootUrl')
+
+        if rootUrl and not o.get('version'):
```

Comment:
Unfortunately, I got the meaning of `version` in references wrong in the node implementation of this change. I'm going to have to go back and fix it but we may as well get it right here from the get-go. Apparently the `version` in a reference is just the version of the reference file format. This is discussed further in https://bugzilla.mozilla.org/show_bug.cgi?id=1460019